src/signalrclient/default_websocket_client.cpp

```diff
@@ -97,14 +99,20 @@ namespace signalr
             {
                 try
                 {
-
                     std::string msg;
                     auto response = task.get();
                     if (response.message_type() == web::websockets::client::websocket_message_type::binary_message)
                     {
-                        throw signalr_exception("binary isn't supported currently");
+                        concurrency::streams::producer_consumer_buffer<uint8_t> b;
+                        response.body().read_to_end(b).get();
+                        auto t = b.create_ostream().streambuf().in_avail();
+                        msg.resize(t, ' ');
```

Comment:
?

Reply:
Need a buffer of size `t`, I don't think you can set a string's size with uninitialized data, so need to add some value, in this case space...

Reply:
You can reserve space with `reserve()` if you just want to preallocate the capacity?

Reply:
I know, but we pass the address of the array directly afterwards and then assume the string is usable. If we used reserve it would fill it up, but the size wouldn't be set.